js_modules/dagster-ui/packages/ui-core/src/app/Util.tsx

```diff
@@ -145,82 +145,84 @@ export function asyncMemoize<T, R, U extends (arg: T, ...rest: any[]) => Promise
   }) as any;
 }
 
-export function indexedDBAsyncMemoize<R, U extends (...args: any[]) => Promise<R>>(
-  fn: U,
-  hashFn?: (...args: Parameters<U>) => any,
-  key?: string,
-): U & {
-  isCached: (...args: Parameters<U>) => Promise<boolean>;
-} {
-  let lru: ReturnType<typeof cache<R>> | undefined;
-  try {
-    lru = cache<R>({
-      dbName: `indexDBAsyncMemoizeDB${key}`,
-      maxCount: 50,
+export const indexedDBAsyncMemoize = weakMapMemoize(
```

Comment:
No change to the content of the function, it's all whitespace

Reply:
This looks good to me, but if we're wrapping this in memoization, is it really necessary for it to internally use `genHashKey = weakMapMemoize(...`? It seems like that second layer of memoization may not really be achieving anything because it is memoizing on the same arguments

Reply:
Oops, not that I look at this, that is being memoized... the expensive part is fetching it from IndexedDB (again). You're right we don't need the inner one with this one in place.